cmd/find-main.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Minio Client (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package cmd stores all of the mc utilities
+package cmd
+
+import (
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+// find specific flags
+var (
+	findFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "Find object names matching wildcard pattern",
+		},
+		cli.StringFlag{
+			Name:  "path",
+			Usage: "Match diretcory names matching wildcard pattern",
+		},
+		cli.StringFlag{
+			Name:  "regex",
+			Usage: "Match directory and object name with PCRE regex pattern",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Print in custom format to STDOUT (see FORMAT)",
+		},
+		cli.StringFlag{
+			Name:  "exec",
+			Usage: "Spawn an external process for each matching object (see FORMAT)",
+		},
+		cli.StringFlag{
+			Name:  "ignore",
+			Usage: "Exclude objects matching the wildcard pattern",
+		},
+		cli.StringFlag{
+			Name:  "newer",
+			Usage: "Match all objects newer than specified time in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "older",
+			Usage: "Match all objects older than specified time in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "larger",
+			Usage: "Match all objects larger than specified size in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "smaller",
+			Usage: "Match all objects smaller than specified size in units (see UNITS)",
+		},
+		cli.StringFlag{
+			Name:  "maxdepth",
+			Usage: "Limit directory navigation to specified depth",
+		},
+		cli.BoolFlag{
+			Name:  "watch",
+			Usage: "Monitor specified location for newly created objects",
+		},
+		cli.BoolFlag{
+			Name:  "or",
+			Usage: "Changes the matching criteria from an \"and\" to an \"or\"",
+		},
+	}
+)
+
+var findCmd = cli.Command{
+	Name:   "find",
+	Usage:  "Finds files which match the given set of parameters.",
+	Action: mainFind,
+	Flags:  append(findFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} PATH FLAG EXPRESSION [FLAG]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+   1. Find all files named foo from all buckets.
+   	$ {{.HelpName}} s3 -name "file"
```

Comment:
I think examples should use the double-dash option format like "--name" to be consistent with examples from other mc commans.